XamlControlsGallery/ControlPagesSampleCode/ListView/ListViewSample4_cs.txt

```diff
@@ -0,0 +1,48 @@
+// C# Code
+
+// Make sure to add this to your declarations at the top for the filtering method:
+using System.Linq;
+// ...
+
+// Define an ObservableCollection<Contact> object to serve as the ListView's
+// ItemsSource. This collection will get updated after the filters are used:
+ObservableCollection<Contact> contactsFiltered = new ObservableCollection<Contact>();
+
+/* Populate contactsFiltered with all contact objects (in this case, contacts holds
+all of our Contact objects so we copy them into contactsFiltered). Set this newly populated
+collection as the ItemsSource for the ListView. */
+contactsFiltered = new ObservableCollection<Contact>(allContacts);
+FilteredListView.ItemsSource = contactsFiltered;
+
+
+// Whenever text changes in any of the filtering text boxes, the following function is called:
+private void OnFilterChanged(object sender, TextChangedEventArgs args)
+{
+    // This is a Linq query that selects only items that return True after being passed through
+    // the Filter function, and adds all of those selected items to filtered.
+    var filtered = allContacts.Where(contact => Filter(contact));
+    Remove_NonMatching(filtered);
+    AddBack_Contacts(filtered);
+}
+
+// The following functions are called inside OnFilterChanged:
+
+/* When the text in any filter is changed, perform a check on each item in the original 
+contact list to see if the item should be displayed, taking into account all three of the
+filters currently applied. If the item passes all three checks for all three filters, 
+the function returns true and the item is added to the filtered list above. */
+private bool Filter(Contact contact)
+{
+    return contact.FirstName.Contains(FilterByFirstName.Text, StringComparison.InvariantCultureIgnoreCase) &&
+            contact.LastName.Contains(FilterByLastName.Text, StringComparison.InvariantCultureIgnoreCase) &&
+            contact.Company.Contains(FilterByCompany.Text, StringComparison.InvariantCultureIgnoreCase);
+}
+
+/* These functions go through the current list being displayed (contactsFiltered), and remove
+any items not in the filtered collection (any items that don't belong), or add back any items 
+from the original allContacts list that are now supposed to be displayed (i.e. when backspace is hit). 
+See XAML Controls Gallery source code for more detail. */
+private void Remove_NonMatching(IEnumerable<Contact> fd) { ... }
+private void AddBack_Contacts(IEnumerable<Contact> fd) { ... }
```

Comment:
I kind of wonder how much value these stubs add? Is there any harm to just writing out all your code here? Feels like you're providing 60% of the implementation and telling people to look at app code for the remaining 40%...why break it up like this?

Reply:
Yeah I was really on the fence about this. I originally asked @stmoy about how to deal with large code segments, and ended up only putting the important chunks in to not clog up the page with code. I felt that the two functions mentioned at the end were more trivial and self-explanatory. However, I'm on the fence so let me know if you think I should just write out all of the code. I think it would add like 15-20 more lines of code.

Reply:
Although the code is trivial to write, it is fundamental for this sample, right? These methods are the specific methods that do the filtering. To that end, my suggestion would be to include the code here.

Reply:
Thanks for all the input. Added back these functions into the code snippet in upcoming commit.